features/messages/impl/src/test/kotlin/io/element/android/features/messages/timeline/factories/event/TimelineItemContentPollFactoryTest.kt

```diff
@@ -0,0 +1,288 @@
+                    answerItems = listOf(
+                        aPollAnswerItem(A_POLL_ANSWER_1).copy(isDisclosed = false, votesCount = 3, percentage = 0.3f),
+                        aPollAnswerItem(A_POLL_ANSWER_2).copy(isDisclosed = false, isSelected = true, votesCount = 6, percentage = 0.6f),
+                        aPollAnswerItem(A_POLL_ANSWER_3).copy(isDisclosed = false, votesCount = 0, percentage = 0f),
+                        aPollAnswerItem(A_POLL_ANSWER_4).copy(isDisclosed = false, votesCount = 1, percentage = 0.1f),
+                    ),
+                    votes = votes,
+                )
+            )
+    }
+
+    @Test
+    fun `Undisclosed poll - ended states`() = runTest {
+        // No votes, no winner
+        Truth.assertThat(
+            factory.create(aPollContent(PollKind.Undisclosed).copy(endTime = 1UL))
+        ).isEqualTo(
+            aTimelineItemPollContent(PollKind.Undisclosed).let {
+                it.copy(
+                    answerItems = it.answerItems.map { answerItem ->
+                        answerItem.copy(isDisclosed = true, isEnabled = false, isWinner = false)
+                    },
+                    isEnded = true,
+                )
+            }
+        )
+
+        // Some votes, according one from current user (winner)
+        var votes = mapOf(
+            A_POLL_ANSWER_1.id to listOf(A_USER_ID_2, A_USER_ID_3, A_USER_ID_4),
+            A_POLL_ANSWER_2.id to listOf(A_USER_ID, A_USER_ID_5, A_USER_ID_6, A_USER_ID_7, A_USER_ID_8, A_USER_ID_9),
+            A_POLL_ANSWER_3.id to emptyList(),
+            A_POLL_ANSWER_4.id to listOf(A_USER_ID_10),
+        )
+        Truth.assertThat(
+            factory.create(aPollContent(PollKind.Undisclosed).copy(votes = votes, endTime = 1UL))
+        )
+            .isEqualTo(
+                aTimelineItemPollContent(PollKind.Undisclosed).copy(
+                    answerItems = listOf(
+                        aPollAnswerItem(A_POLL_ANSWER_1).copy(isDisclosed = true, isEnabled = false, votesCount = 3, percentage = 0.3f),
+                        aPollAnswerItem(A_POLL_ANSWER_2).copy(
+                            isDisclosed = true,
+                            isSelected = true,
+                            isEnabled = false,
+                            isWinner = true,
+                            votesCount = 6,
+                            percentage = 0.6f
+                        ),
+                        aPollAnswerItem(A_POLL_ANSWER_3).copy(isDisclosed = true, isEnabled = false, votesCount = 0, percentage = 0f),
+                        aPollAnswerItem(A_POLL_ANSWER_4).copy(isDisclosed = true, isEnabled = false, votesCount = 1, percentage = 0.1f),
+                    ),
+                    votes = votes,
+                    isEnded = true,
+                )
+            )
+
+        // Some votes, according one from current user (not winner) and two winning votes
```

Comment:
General remark, Instead of putting several cases in the same test, maybe split into several distinct tests with long names?

Reply:
Done, I put several cases in the same test because I thought it was easier to distinguish the test cases.